update-stable-ids/Jenkinsfile

```diff
@@ -0,0 +1,122 @@
+import groovy.json.JsonSlurper
+// This Jenkinsfile is used by Jenkins to run the UpdateStableIdentifiers step of Reactome's release.
+// It requires that the ConfirmReleaseConfigs step has been run successfully before it can be run.
+pipeline {
+	agent any
+
+	stages {
+		// This stage checks that an upstream project, ConfirmReleaseConfig, was run successfully for its last build.
+		stage('Check ConfirmReleaseConfig build succeeded'){
+			steps{
+				script{
+					// This queries the Jenkins API to confirm that the most recent build of ConfirmReleaseConfigs was successful.
+					def configStatusUrl = httpRequest authentication: 'jenkinsKey', url: "${env.JENKINS_JOB_URL}/job/${env.RELEASE_NUMBER}/job/ConfirmReleaseConfigs/lastBuild/api/json"
+					def configStatusJson = new JsonSlurper().parseText(configStatusUrl.getContent())
+					if(configStatusJson['result'] != "SUCCESS"){
+						error("Most recent ConfirmReleaseConfigs build status: " + configStatusJson['result'] + ". Please complete a successful build.")
+					}
+				}
+			}
+		}
+		// This stage moves 'slice_current' database to 'slice_previous', and then moves 'slice_test' to 'slice_current'.
+		// It also saves the slice_test dump as a snapshot, to be used in the next release.
+		stage('Setup: Rotate slice DBs'){
+			steps{
+				script{
+					dir('update-stable-ids'){
+						withCredentials([usernamePassword(credentialsId: 'mySQLUsernamePassword', passwordVariable: 'pass', usernameVariable: 'user')]){
+							def slice_test_snapshot_dump = "${env.SLICE_TEST}_${env.RELEASE_NUMBER}_snapshot.dump"
+							sh "mysql -u$user -p$pass -e \'drop database if exists ${env.SLICE_PREVIOUS}; create database ${env.SLICE_PREVIOUS}\'"
+							sh "zcat  archive/${env.PREV_RELEASE_NUMBER}/${env.SLICE_TEST}_${env.PREV_RELEASE_NUMBER}_snapshot.dump.gz 2>&1 | mysql -u$user -p$pass ${env.SLICE_PREVIOUS}"
+							sh "mysqldump -u$user -p$pass ${env.SLICE_TEST} > $slice_test_snapshot_dump"
+							sh "gzip -f $slice_test_snapshot_dump"
+							sh "mysql -u$user -p$pass -e \'drop database if exists ${env.SLICE_CURRENT}; create database ${env.SLICE_CURRENT}\'"
+							sh "zcat  ${env.SLICE_TEST}_${env.RELEASE_NUMBER}_snapshot.dump.gz 2>&1 | mysql -u$user -p$pass ${env.SLICE_CURRENT}"
+						}
+					}
+				}
+			}
+		}
+		// This stage backs up the gk_central database before it is modified.
+		stage('Setup: Back up gk_central'){
```

Comment:
Is the database still named "gk_central" on the server, or has it be renamed to "central"? I think that keeping the "gk_" here could cause future confusion with other developers, if they don't know that "gk_" is a historic prefix, not used anywhere else anymore.  BUT... if the database is actually still using the name "gk_central" in MySQL, then it's probably OK to leave this as-is.

Reply:
This is actually gk_central that's found on the curator server. On release we've changed it to be named 'central', but not on curator. I could generalize it to be named something like 'Back up curator database' if we think that would be better for others

Reply:
No, that's OK. I didn't realize it was still "gk_central" on the Curator server.